front/pages/api/user/onboarding-complete.ts

```diff
@@ -0,0 +1,128 @@
+import { isLeft } from "fp-ts/lib/Either";
+import * as t from "io-ts";
+import * as reporter from "io-ts-reporters";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { withSessionAuthentication } from "@app/lib/api/auth_wrappers";
+import type { SessionWithUser } from "@app/lib/iam/provider";
+import { getUserFromSession } from "@app/lib/iam/session";
+import { UserResource } from "@app/lib/resources/user_resource";
+import { ServerSideTracking } from "@app/lib/tracking/server";
+import { renderLightWorkspaceType } from "@app/lib/workspace";
+import logger from "@app/logger/logger";
+import { apiError } from "@app/logger/withlogging";
+import type { WithAPIErrorResponse } from "@app/types";
+
+export type PostOnboardingCompleteResponseBody = {
+  success: boolean;
+};
+
+const PostOnboardingCompleteBodySchema = t.type({
+  metadata: t.record(t.string, t.union([t.string, t.number, t.boolean])),
+});
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<
+    WithAPIErrorResponse<PostOnboardingCompleteResponseBody>
+  >,
+  session: SessionWithUser
+): Promise<void> {
+  if (req.method !== "POST") {
+    return apiError(req, res, {
+      status_code: 405,
+      api_error: {
+        type: "method_not_supported_error",
+        message: "The method passed is not supported, POST is expected.",
+      },
+    });
+  }
+
+  const bodyValidation = PostOnboardingCompleteBodySchema.decode(req.body);
+  if (isLeft(bodyValidation)) {
+    const pathError = reporter.formatValidationErrors(bodyValidation.left);
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: `Invalid request body: ${pathError}`,
+      },
+    });
+  }
+
+  const user = await getUserFromSession(session);
+  if (!user) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "user_not_found",
+        message: "The user was not found.",
+      },
+    });
+  }
+
+  const u = await UserResource.fetchByModelId(user.id);
+  if (!u) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "user_not_found",
+        message: "Could not find the user.",
+      },
+    });
+  }
+
+  // Get the first workspace (assuming onboarding is for the primary workspace)
+  const workspace = user.workspaces[0];
+  if (!workspace) {
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "User has no associated workspace.",
+      },
+    });
+  }
+
+  const { metadata } = bodyValidation.right;
```

Comment:
Maybe we should rather just pass "jobType" instead of arbitrary metadata ? That would make the code more obvious (completing onboarding means the user inputs specific fields, and we make it explicit which ones)